### PR TITLE
Add t_cdf, t_distribution scalar functions

### PR DESCRIPTION
**Pull request by wendigo**
https://support.microsoft.com/en-us/office/t-dist-function-4329459f-ae91-48c2-bba8-1ead1c6c21b2

### CODE REVIEW

**Comment by martint** [comment]
The cumulative version should follow the other cumulative distribution functions (see https://trino.io/docs/current/functions/math.html#cumulative-distribution-functions):

```
t_cdf(x, df)
```

Let me think about the probability density function form. Maybe `t_pdf` or `t_distribution`.

File: core/trino-main/src/test/java/io/trino/operator/scalar/TestDistributionFunctions.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import static io.trino.testing.assertions.TrinoExceptionAssert.assertTrinoExceptionThrownBy;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+public class TestDistributionFunctions
+{
+    private QueryAssertions assertions;
```
Comment (martint) [inline core/trino-main/src/test/java/io/trino/operator/scalar/TestDistributionFunctions.java:32]: Make this final and initialize it here.

Comment (wendigo) [reply core/trino-main/src/test/java/io/trino/operator/scalar/TestDistributionFunctions.java]: I've moved tests to TestMathFunctions

**Comment by electrum** [comment]
Can you add these to the documentation? We keep the documentation in the same source tree so that we can update it when the functionality is added.

**Comment by wendigo** [comment]
@electrum done

File: core/trino-main/src/main/java/io/trino/operator/scalar/MathFunctions.java [hunk 2]
```diff
@@ -702,6 +703,32 @@ public static double betaCdf(
...
+            @SqlType(StandardTypes.DOUBLE) double degreesOfFreedom)
+    {
+        if (degreesOfFreedom < 1) {
+            throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "degrees of freedom must be greater than or equal to 1");
+        }
+        return new TDistribution(degreesOfFreedom).cumulativeProbability(x);
+    }
+
+    @Description("Students' t-distribution given the x and degrees of freedom")
+    @ScalarFunction(alias = "t_pdf")
```
Comment (martint) [inline core/trino-main/src/main/java/io/trino/operator/scalar/MathFunctions.java:720]: Let's just got with `t_pdf`. "distribution" is a broader concept, and pdf/cdf describe certain properties of a distribution. In particular, pdf describes density of probability of a random variable at a given point for the distribution.

File: core/trino-main/src/main/java/io/trino/operator/scalar/MathFunctions.java [hunk 3]
```diff
@@ -702,6 +703,32 @@ public static double betaCdf(
         return distribution.cumulativeProbability(value);
     }
 
+    @Description("Students' t-distribution cdf given the x and degrees of freedom")
```
Comment (martint) [inline core/trino-main/src/main/java/io/trino/operator/scalar/MathFunctions.java:706]: > *Student's* t-distribution ...

File: docs/src/main/sphinx/functions/math.md [hunk 4]
```diff
@@ -271,3 +277,8 @@ Compute the Normal cdf with given mean and standard deviation (sd):  P(N \< v; m
 The mean and value v must be real values and the standard deviation must be a real
 and positive value.
 :::
+
+:::{function} t_cdf(x, df) -> double
+Compute the Student's t-distribution cdf with given x and degrees of freedom (df).
```
Comment (martint) [inline docs/src/main/sphinx/functions/math.md:282]: Similarly, replace "cdf" with "cumulative density"

**Pull request merged**